src/check_release.rs

```diff
@@ -230,6 +231,46 @@ fn print_triggered_lint(
     Ok(())
 }
 
+/// Helper function to map a given semver query, and its witness to a value for use in [`run_check_release`] to
+/// to an [`Option`].
+fn map_witness_queries<'query>(
+    semver_query: &'query SemverQuery,
+    lint_results: &Vec<BTreeMap<Arc<str>, FieldValue>>,
+    adapter: &VersionedRustdocAdapter,
+) -> Option<(
+    &'query SemverQuery,
+    Vec<anyhow::Result<BTreeMap<Arc<str>, FieldValue>>>,
```

Comment:
I'm not entirely sold on returning this `Vec` as opposed to immediately acting on the results.

Consider writing out more of the witness logic (expanding beyond the original scope of the PR) so we can see what fits best?